### DISCUSSION TITLE
Should server send multiple Task Finish notification after request cancellation?

### DISCUSSION THREAD

**Discussion by PBundyra** [build-server-protocol/build-server-protocol#515] [Q&A]
Let's say we have three levels in the tasks hierarchy - main task, subtask, and sub-subtask. Should the server send three Task Finish notifications with the Cancelled Status Code, each corresponding to one level in the hierarchy, or maybe status of the task is inherited by default, and the server should send only one Task Finish notification corresponding to the main task?

**Comment by jastice** [comment]
I would say the server *should* send a `Cancelled` code for each task it sent a `TaskStarted` notification, but the client should also be permissive of incomplete implementations and assume subtasks are cancelled if a parent was cancelled.
